dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k.cc

```diff
@@ -240,18 +226,28 @@ DecodeResult NvJpeg2000DecoderInstance::DecodeImplTask(int thread_idx,
 
   CUDA_CALL(cudaEventSynchronize(ctx.decode_event));
 
-  // nvJPEG2000 decodes into planar layout
-  bool is_conversion_needed = ctx.shape[2] > 1 || ctx.bpp == 16;
+  DALIImageType format = ctx.shape[2] == 1 ? DALI_GRAY : DALI_RGB;
+  bool is_processing_needed =
+    ctx.shape[2] > 1 ||  // nvJPEG2000 decodes into planar layout
+    ctx.pixel_type != opts.dtype ||
+    format != opts.format ||
+    (ctx.bpp != 8 && ctx.bpp != 16);
+
   try {
-    if (!is_conversion_needed) {
-      result.success = DecodeJpeg2000(in, out.mutable_data<uint8_t>(), ctx);
-    } else {
-      auto &buffer = res.intermediate_buffer;
-      buffer.clear();
+    auto decode_out = out;
+    if (is_processing_needed) {
       int64_t pixel_size = dali::TypeTable::GetTypeInfo(ctx.pixel_type).size();
```

Comment:
I would expect the pixel to contain all the color values, it is a bit misleading name

Reply:
What would be a better name there? I don't really know the terminology.